src/modules/tools/spindle/SoftSerial/SoftSerial_rx.cpp

```diff
@@ -0,0 +1,91 @@
+/* 
+ * SoftSerial by Erik Olieman
+ * Date: 05 Jul 2014
+ * Revision: 10:236fce2e5b8c
+ * URL: http://developer.mbed.org/users/Sissors/code/SoftSerial/
+ */
+
+#include "SoftSerial.h"
+
+uint32_t overhead_us = 200 * 1000000 / SystemCoreClock;         //Random estimation of the overhead of mbed libs, makes slow devices like LPC812 @ 12MHz perform better
+
+int SoftSerial::_getc( void ) {
+    while(!readable());

```

Comment:
it really is not acceptable to have a loop like this that stops anything from running. ON_IDLE needs to be called at least.